src/v/config/property.h

```diff
@@ -911,4 +956,43 @@ class retention_duration_property final
     }
 };
 
+template<typename T>
+class optional_property final : public property<std::optional<T>> {
+public:
+    optional_property(
+      config_store& conf,
+      std::string_view name,
+      std::string_view desc,
+      base_property::metadata meta = {},
+      std::optional<T> def = T{},
+      property<std::optional<T>>::validator validator
+      = property<std::optional<T>>::noop_validator,
+      std::optional<legacy_default<std::optional<T>>> ld = std::nullopt,
+      make_constraint_methods_h constraint_methods_h
+      = property<std::optional<T>>::make_default_constraint_methods)
+      : property<std::optional<T>>(
+        conf,
+        std::move(name),
+        std::move(desc),
+        std::move(meta),
+        std::move(def),
+        std::move(validator),
+        std::move(ld),
+        std::move(constraint_methods_h)) {}
+
+    template<typename Func>
+    helpers::range_values<T> make_range(
+      const std::optional<ss::sstring>& min_str,
+      const std::optional<ss::sstring>& max_str,
+      Func&& type_converter) {
+        auto min = min_str ? std::make_optional(
+                     static_cast<T>(type_converter(*min_str)))
+                           : std::nullopt;
+        auto max = max_str ? std::make_optional(
+                     static_cast<T>(type_converter(*max_str)))
+                           : std::nullopt;
+        return helpers::range_values<T>{std::move(min), std::move(max)};
+    }
```

Comment:
This looks almost the same as the one in property, which isn't virtual and doesn't rely on any state. It makes me wonder if `make_range` shouldn't just be a free function that can be used from the 3 places it's implemented.

Reply:
`range_values` needs the integral type and the type for each config is captured in the property class that is why `make_range` is a member of property classes.

Reply:
This method will be irrelevant in the next push following a change to integral properties.